omisego-sdk/src/main/java/co/omisego/omisego/websocket/interval/SocketHeartbeat.kt

```diff
@@ -38,19 +39,26 @@ class SocketHeartbeat(
      * @param task A lambda with a [SocketSend] parameter. This will be executed periodically that starts immediately.
      */
     @Suppress("OVERRIDE_BY_INLINE")
+    @Synchronized
     override inline fun startInterval(crossinline task: (SocketSend) -> Unit) {
-        timer = Timer()
-        timer?.schedule(Date(), period, {
-            task(SocketSend(EVENT_NAME, SocketEventSend.HEARTBEAT, socketMessageRef.value, mapOf()))
-        })
+        synchronized(this) {
+            timer?.cancel()
+            timer = Timer()
+            timer?.schedule(Date(), period, {
```

Comment:
Not to be picky, but is there a version with only long in the kotlin lib? like this one
https://docs.oracle.com/javase/7/docs/api/java/util/Timer.html#schedule(java.util.TimerTask,%20long,%20long)

This would avoid creating an unnecessary date object 😄